source/simulator/assembly.cc

```diff
@@ -1624,14 +1644,37 @@ namespace aspect
     // TODO: Compute artificial viscosity once per timestep instead of each time
     // temperature system is assembled (as this might happen more than once per
     // timestep for iterative solvers)
-    const double nu
-      = compute_viscosity (scratch,
+
+    double nu = 0;
+	int count_additions = 0;
+	double nu_additions = 0.0;
+    if (use_smoothing == false)
+    {
+      nu = compute_viscosity (scratch,
```

Comment:
I think this is awkward. Just compute the viscosity outside this function somewhere and store it in the vector. If necessary smooth the elements of the vector. Then pass this vector into this function and use it -- at this point you then no longer have to care about smoothing or not.